exasol/ds/sandbox/lib/setup_ec2/run_setup_ec2.py

```diff
@@ -123,54 +166,56 @@ def run_setup_ec2(
     ec2_key_name: Optional[str],
     asset_id: AssetId,
     configuration: ConfigObject,
+    dependency_installer: Optional[AnsibleDependencyInstaller],
 ) -> None:
     """
     Launches an EC2-instance and then waits until the user presses Ctrl-C,
     then shuts down the instance again.
 
     :param aws_access: AWSAccess proxy.
+    :param ec2_instance_type: The name of the EC2 instance type to use,
+           e.g. "t2.medium".
     :param ec2_key_file: The private key file to use for the EC2-Instance.
     :param ec2_key_name: The key name of the key to use for the EC2-Instance.
     :param asset_id: The asset id to use: Will use the tags (for the
            cloudformation stack and the key) and the prefix of the
            cloudformation stack
     :param configuration: The global configuration to use.
-    :param ec2_instance_type: The name of the EC2 instance type to use,
-           e.g. "t2.medium".
+    :param dependency_installer: If provided then additionally install
+           dependencies using the provided installer.
     """
-    source_ami_id = ec2_source_ami or source_ami_id_with_logging(
-        aws_access,
-        configuration.source_ami_filters,
-        LOG,
+    finder = AmiFinder(aws_access, LOG)
+    source_ami = (
+        finder.unique({"image-id": ec2_source_ami}) if ec2_source_ami
+        else finder.latest(configuration.source_ami_filters)
     )
+    if not source_ami:
+        return
     execution_generator = run_lifecycle_for_ec2(
         aws_access=aws_access,
         ec2_instance_type=ec2_instance_type,
         ec2_key_file=ec2_key_file,
         ec2_key_name=ec2_key_name,
         asset_id=asset_id,
-        ami_id=source_ami_id,
+        ami_id=source_ami.id,
         user_name=None,
     )
     with EC2StackLifecycleContextManager(execution_generator, configuration) as res:
         ec2_instance: Optional[EC2Instance]
         key_file_location: Optional[str]
         ec2_instance, key_file_location = res
-
-        if not ec2_instance.is_running:
-            LOG.error(f"Error during startup of EC2 instance "
-                      f"'{ec2_instance.id}'. "
-                      f"Status is {ec2_instance.state_name}")
-        else:
-            LOG.info(f"You can now login to the ec2 machine with "
-                     f"'ssh -i {key_file_location} "
-                     f"ubuntu@{ec2_instance.public_dns_name}'")
-        LOG.info('Press Ctrl+C to stop and cleanup.')
+        _install_dependencies_and_report_status(
```

Comment:
Call `_install_dependencies_and_report_status()` only if variable `dependency_installer` is not None.

Reply:
This is a very old project, not using the PTB.